packages/utils/src/baggage.ts

```diff
@@ -0,0 +1,54 @@
+export type AllowedBaggageKeys = 'environment' | 'release'; // TODO: Add remaining allowed baggage keys | 'transaction' | 'userid' | 'usersegment';
+export type Baggage = Partial<Record<AllowedBaggageKeys, string> & Record<string, string>>;
+
+export const BAGGAGE_HEADER_NAME = 'baggage';
+
+/**
+ * Max length of a serialized baggage string
+ *
+ * https://www.w3.org/TR/baggage/#limits
+ */
+export const MAX_BAGGAGE_STRING_LENGTH = 8192;
+
+/** Create an instance of Baggage */
+export function createBaggage(initItems: Baggage): Baggage {
+  return { ...initItems };
+}
+
+/** Add a value to baggage */
+export function getBaggageValue(baggage: Baggage, key: keyof Baggage): Baggage[keyof Baggage] {
+  return baggage[key];
+}
+
+/** Add a value to baggage */
+export function setBaggageValue(baggage: Baggage, key: keyof Baggage, value: Baggage[keyof Baggage]): void {
+  baggage[key] = value;
+}
+
+/** remove a value from baggage */
+// TODO: Is this even useful?
+// export function removeBaggageValue(baggage: Baggage, key: keyof Baggage): void {
+//   // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+//   delete baggage[key];
+// }
+
+/** Serialize a baggage object */
+export function serializeBaggage(baggage: Baggage): string {
+  return Object.keys(baggage).reduce((prev, key) => {
+    const newVal = `${prev};${key}=${baggage[key as keyof Baggage]}`;
+    return newVal.length > MAX_BAGGAGE_STRING_LENGTH ? prev : newVal;
+  }, '');
+}
+
+/** Parse a baggage header to a string */
+export function parseBaggageString(baggageString: string): Baggage {
+  // TODO: Should we check validity with regex? How much should we check for malformed baggage keys?
+  // Perhaps we shouldn't worry about this being used in the frontend, so bundle size isn't that much of a concern
```

Comment:
Just to confirm what you mean with the comment: We could make more extensive baggage validity checks because the `parseBaggageString` function won't be used in Browser environments and is thus tree-shaken out, which means the checks wouldn't hurt bundle size?

Reply:
Yes exactly that, I can update my comment to make it a little more clear.

Reply:
I think that'd be great. I was just asking myself if this is actually the case. Is there a situation in which we have to parse the header if other services already created it and put stuff into it? Or does this not apply for browser environments?